lib/ui/painting.dart

```diff
@@ -1684,21 +1681,34 @@ class Codec extends NativeFieldWrapperClass2 {
   /// * -1 for infinity repetitions.
   int get repetitionCount native 'Codec_repetitionCount';
 
+  FrameInfo _cachedFrame;
+
   /// Fetches the next animation frame.
   ///
   /// Wraps back to the first frame after returning the last frame.
   ///
   /// The returned future can complete with an error if the decoding has failed.
-  Future<FrameInfo> getNextFrame() {
-    return _futurize(_getNextFrame);
+  Future<FrameInfo> getNextFrame() async {
+    if (_cachedFrame != null && frameCount == 1) {
+      return _cachedFrame;
+    }
+    final Image image = Image._();
+    print('About to call _futureize with image');
```

Comment:
Did you mean to leave this debug logging here?

Reply:
Removed.  Thanks for catching this, this would've caused actual confusion for users haha